odoo/addons/base/module/module.py

```diff
@@ -463,7 +463,8 @@ def module_uninstall(self):
         """
         modules_to_remove = self.mapped('name')
         self.env['ir.model.data']._module_data_uninstall(modules_to_remove)
-        self.write({'state': 'uninstalled', 'latest_version': False})
+        # we deactivate prefetching to not try to read a column that has been deleted
+        self.with_context({'prefetch_fields': False}).write({'state': 'uninstalled', 'latest_version': False})
```

Comment:
I would not discard the whole context here:
```suggestion
        self.with_context(prefetch_fields=False).write({'state': 'uninstalled', 'latest_version': False})
```

Reply:
Of course, thanks.